pkg/plugin/processor/builtin/impl/webhook/http.go

```diff
@@ -72,6 +74,27 @@ type httpConfig struct {
 	ResponseStatusRef string `json:"response.status"`
 }
 
+func (c *httpConfig) validateHeaders() error {
+	if c.Headers == nil {
+		c.Headers = make(map[string]string)
+	}
+
+	var headerValue string
+	for name, value := range c.Headers {
+		if strings.ToLower(name) == "content-type" {
+			headerValue = value
+		}
+	}
+
+	if headerValue != "" && c.ContentType != "" {
+		return cerrors.Errorf("Configuration error, cannot provide both \"request.contentType\" and \"headers.Content-Type\", use \"headers.Content-Type\" only.")
+	}
+
+	c.Headers["Content-Type"] = headerValue
```

Comment:
We still have 2 issues - we might be duplicating the `Content-Type` header if it uses a different case (e.g. `content-type`), and we are not using the deprecated `ContentType` option.

Reply:
Yup, I'm aware of that, but I wasn't sure why the tests don't show that. Dumb thing, I was checking errors only, not the configs too.:)